components/cdc/src/endpoint.rs

```diff
@@ -306,6 +319,47 @@ impl ResolvedRegionHeap {
         self.clear();
         self.heap.shrink_to(min_capacity);
     }
+
+    // extreme outier match the following two conditions:
+    // 1. https://en.wikipedia.org/wiki/Box_plot
+    // 2. the gap with min_ts is larger than RAW_RESOLVED_TS_OUTLIER_THRESHOLD.
+    fn get_extreme_outlier(&self, min_ts: TimeStamp) -> Vec<u64> {
+        if self.heap.is_empty() {
+            return vec![];
+        } else if self.heap.len() < 4 {
+            // if count is smaller than 4, just return one outlier if exists.
```

Comment:
I suggest not to detect the outlier when number of regions less than 4 or a threshold (maybe 10 ?). When the number is small, the confidence of outlier detection is low.
As there is no good method to do so now, just don't do it.
Moreover, this scenario should not happen in a production environment.

Reply:
ok, just detect when region number is more than 10.